### ISSUE TITLE
Jedi error: Cannot call write after a stream was destroyed

### ISSUE THREAD

**Issue by arishabh** [neoclide/coc-python#217]
I am getting this error after I installed coc-python. Any help?

**Comment by chemzqm** [comment]
`:messages`, you may need upgrade your jedi.

**Comment by arishabh** [comment]
`
[coc.nvim] Jedi error: Cannot call write after a stream was destroyed
[coc.nvim] Jedi error: Cannot call write after a stream was destroyed
[coc.nvim] Jedi error: Cannot call write after a stream was destroyed
[coc.nvim] error: Some plugin change completeopt on insert mode: Last set from ~/.vim/plugged/jedi-vim/autoload/jedi.vim line 716!
[coc.nvim] error: Some plugin change completeopt on insert mode: Last set from ~/.vim/plugged/jedi-vim/autoload/jedi.vim line 716!
[coc.nvim] error: Some plugin change completeopt on insert mode: Last set from ~/.vim/plugged/jedi-vim/autoload/jedi.vim line 716!
[coc.nvim] error: Some plugin change completeopt on insert mode: Last set from ~/.vim/plugged/jedi-vim/autoload/jedi.vim line 716!
[coc.nvim] Jedi error: Cannot call write after a stream was destroyed
`

This is what I am getting. The thing is I don't have a jedi-vim dir in my plugged dir.

**Comment by chemzqm** [comment]
You should not use jedi-vim and coc.nvim for completion at the same time.

**Comment by arishabh** [comment]
Sorry, this is the actual error I am getting

`Jedi error: Traceback (most recent call last):
[coc.nvim] Jedi error:     import jedi
[coc.nvim] Jedi error: Cannot call write after a stream was destroyed
[coc.nvim] Jedi error: Cannot call write after a stream was destroyed
`

**Comment by arishabh** [comment]
Please help me out!!

**Comment by chemzqm** [comment]
Looks like jedi is not found

**Comment by arishabh** [comment]
Well, do I need to download anything else?

**Comment by arishabh** [comment]
Hey, any ideas?

**Comment by kerrigoon** [comment]
I think you need to download jedi as well.

I created a separate environment for jedi, autopep8 and pylint and then added:
"python.formatting.autopep8Path": "/path/to/env/bin/autopep8",
"python.jediPath": "/path/to/env/lib/python-version/site-packages/",
"python.sortImports.path": "/path/to/env/bin/isort",
to my coc-settings.json and is working fairly well. I get a few timeout errors but on the whole it is fine. 

Pylint seems to be needed in every environment you want to work in as if you supply a path to it only the modules in the environment of that path are picked up. I can't seem to find a way to tell it which rope to use so you have to have that installed in each environment as well it would appear. 

I think there is a way to do this by pointing to a folder with virtual environments stored in them but I don't know how that works or what that setting is really for. Similarly a few issues mention people installing the neovim python package in every environment but mine seems to be working fine without it, but again I'm not sure what that does. 

Hope this helps

**Comment by arishabh** [comment]
I am sure it will! Thank you

**Comment by JonasRSV** [comment]
I have this issue was well. Did you manage to resolve it?

**Comment by arishabh** [comment]
No, I just don't use jedi now lol. Just coc

**Comment by GianniGi** [comment]
I use coc.nvim, yet I get the jedi error.... why????

**Comment by tedmiddleton** [comment]
Yeah this is really frustrating. coc-python seems to be completely broken. I'm using python 3.8 on Fedora, I *do* have jedi installed as part of yum/dnf, but I never really explicitly opted in to it - to me its just an implementation detail of a broken coc plugin?

**Comment by GianniGi** [comment]
I've solved by upgrading python to 3.8. Something. The latest anyway

Il lun 7 set 2020, 02:52 Ted Middleton <notifications@github.com> ha
scritto:

**Comment by pavlosun** [comment]
Same issue. Solved by setting `python.jediPath` and reinstalling `coc-python`. During `coc-python` install got the following error:
```
[coc.nvim] Error on load extension from "/home/<username>/.config/coc/extensions/node_modules/coc-python": item.dispose is not a function
```
But it is probably a different story.
